src/Compilers/Core/Portable/CodeGen/ILBuilder.cs

```diff
@@ -70,7 +70,7 @@ internal ILBuilder(ITokenDeferral module, LocalSlotManager localSlotManager, Opt
 
             leaderBlock = _currentBlock = _scopeManager.CreateBlock(this);
 
-            _labelInfos = new SmallDictionary<object, LabelInfo>(ReferenceEqualityComparer.Instance);
+            _labelInfos = new SmallDictionary<object, LabelInfo>();
```

Comment:
The behavior change is intentional. Label declarations and references are distinct instances when binding global statements. FWIW, ```PreciseAbstractFlowPass._labels``` uses default EqualityComparer.